test/core/memory_usage/memstats.h

```diff
@@ -18,15 +18,53 @@
 #include <stdlib.h>
 #include <sys/resource.h>
 
+#include <fstream>
+#include <iostream>
+
+#include "absl/strings/str_cat.h"
+
 // IWYU pragma: no_include <bits/types/struct_rusage.h>
 
+// Get the memory usage of either the calling process or another process using
+// the pid
+long GetMemUsage(absl::optional<int> pid = absl::nullopt) {
```

Comment:
free functions in headers need to be inline or they'll produce ODR violations, however this function is probably larger than I'd like to see inlined... soo let's leave a function prototype here and move the implementation into memstats.cc.

Reply:
I see, moved the implementation into memstats.cc